server/db/actions/Dog.js

```diff
@@ -0,0 +1,59 @@
+import dbConnect from "../dbConnect";
+import Dog from "../models/Dog";
+import Log from "../models/Log";
+import User from "../models/User";
+
+/**
+ * Updates an existing dog in the database
+ * @param {*} dogId ID of the Dog Object to be updated
+ * @param {*} dogData Dog Object that has been parsed by Zod for validity
+ * @returns the Dog Object prior to the update; error otherwise
+ */
+export async function updateDog(dogId, dogData) {
+  await dbConnect();
```

Comment:
Now that the last catch is not a static message, if `await dbConnect()` throws it will send a dev-specific error to the client, which we don't want. We can wrap this in a try/catch and throw an error saying `"Unable to update dog, please try again"`

Reply:
done :)